### PR TITLE
Make it possible to configure in-line composites

### PR DESCRIPTION
**Pull request by pnuu**
With this PR it will be possible to define composites in-line inside of another composites. This will deprecate several composites currently using `Dust` and `Airmass` compositors.

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master -- "*py" | flake8 --diff`` 
 - [ ] Fully documented

### CODE REVIEW

File: satpy/composites/__init__.py [hunk 1]
```diff
@@ -154,10 +153,21 @@ def _process_composite_config(self, composite_name, conf,
...
-                    # 'modifiers' being None otherwise it will be an empty
-                    # tuple
-                    item.setdefault('modifiers', None)
+                    # Handle in-line composites
+                    if 'compositor' in item:
+                        # Create an unique temporary name for the composite
+                        import uuid
+                        temp_comp_name = 'temp_' + str(uuid.uuid4())
+                        # Minimal composite config
+                        conf_tmp = {composite_type: {temp_comp_name: item}}
```
Comment (djhoese) [inline satpy/composites/__init__.py]: Would it be possible to hash the composite arguments or something so it isn't unique? This way composites could be reused...eh maybe that's a bad idea. Actually that's the whole point of defining the dependencies not inline.

**Comment by djhoese** [comment]
To be clear and document this could you show me what one of these inline compositors looks like?

**Comment by pnuu** [comment]
I think I'm now finished with this, unless there are good suggestions. I didn't update the composite documentation, as it is basically missing and will require a full issue of its own.

**Comment by djhoese** [comment]
My version of this PR shows that there is a conflict with compositor_tests? Probably from @mraspaud recent MODIS thin PR.

**Comment by pnuu** [comment]
Thanks for pointing that out, rebased to current master.

**Comment by djhoese** [comment]
@pnuu Is this ready to go? I know you asked me to do some performance testing, but other than that is this done?

**Comment by djhoese** [comment]
Tests for these situations should probably be added to the `test_scene.py` module or somewhere similar, but these tests are admittedly out of control and complex.

**Comment by pnuu** [comment]
I added a test that verifies the instrument specific composite overrides the one from `visir.yaml` also for the in-lined composites (the sub-composite prerequisites are using channel names instead of wavelengths).

**Pull request merged**